homeassistant/components/shelly/__init__.py

```diff
@@ -85,6 +89,9 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         )
         return False
 
+    if entry.data.get("gen") == 2:
```

Comment:
I think that we shouldn't store it in the data because a firmware upgrade could bump the available API? In that case we should detect the generation using `get_info` maybe?

Reply:
btw using numbers for `gen` but then having all other references be `block` or `rpc` is a bit confusing. Should we drop `1` and `2` for `gen` ?

Reply:
> I think that we shouldn't store it in the data because a firmware upgrade could bump the available API? In that case we should detect the generation using `get_info` maybe?

This was requested to be added by us to indicate the device protocol type (Shelly calls it generation) so that 1 is CoAP and 2 is RPC. The main problem is that for CoAP we also support sleeping devices and can't query `get_info` at startup.
I do hope they won't use that to indicate API version, but if they do we can change our logic to query `get_info` if we detect it is an RPC device (so that we don't do that for CoAP based devices).

Reply:
> btw using numbers for `gen` but then having all other references be `block` or `rpc` is a bit confusing. Should we drop `1` and `2` for `gen` ?

Agree, I will think of a better way and address it in the next PR